packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -286,35 +283,66 @@ describe("Garbage Collection Tests", () => {
 
             await garbageCollector.collectGarbage({ runGC: true });
 
-            // Update all nodes.
-            updateAllNodes(garbageCollector);
-
-            // Validate that no inactive events are generated yet.
+            // Validate that no inactive events are logged yet.
+            await updateAllNodesAndRunGC(garbageCollector);
             validateNoInactiveEvents();
 
             // Expire the unreferenced timer (if any).
             clock.tick(inactiveTimeoutMs + 1);
 
-            // Update all nodes. This should result in an inactiveObjectChanged event for node 2 and node 3 since they
-            // are inactive.
-            updateAllNodes(garbageCollector);
+            // Validate that changed and loaded events for node 2 and node 3 are logged since they are inactive.
+            await updateAllNodesAndRunGC(garbageCollector);
             assert(
                 mockLogger.matchEvents([
                     { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[2], pkg: eventPkg },
                     { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[2], pkg: eventPkg },
                     { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
                     { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
                 ]),
-                "inactive events not generated as expected",
+                "inactive events not logged as expected",
             );
 
-            // Add reference from node 1 to node 3 and validate that we get revivedEvent event.
+            // Add reference from node 1 to node 3 and validate that revived event is logged.
             garbageCollector.addedOutboundReference(nodes[1], nodes[3]);
+            await garbageCollector.collectGarbage({});
             assert(
                 mockLogger.matchEvents([
-                    { eventName: revivedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
+                    { eventName: revivedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg, fromId: nodes[1] },
                 ]),
-                "inactive event not generated as expected",
+                "revived event not logged as expected",
+            );
+        });
+
+        it("generates only revived event when an inactive node is changed and revived", async () => {
+            const garbageCollector = createGarbageCollector();
+
+            // Remove node 2's reference from node 1. This should make node 2 and node 3 unreferenced.
+            defaultGCData.gcNodes[nodes[1]] = [];
+
+            await garbageCollector.collectGarbage({ runGC: true });
+
+            // Validate that no inactive events are logged yet.
+            await updateAllNodesAndRunGC(garbageCollector);
+            validateNoInactiveEvents();
+
+            // Expire the unreferenced timer (if any).
+            clock.tick(inactiveTimeoutMs + 1);
```

Comment:
Should we have a before each and after each resetting the clock here or is it globally done at a higher level?

Reply:
The clock is reset after each test. It's in the afterEach block for the outer most describe.